libnavigation-util/src/test/java/com/mapbox/navigation/utils/internal/TimeTest.kt

```diff
@@ -0,0 +1,29 @@
+package com.mapbox.navigation.utils.internal
+
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import kotlin.math.abs
+
+class TimeTest {
+
+    @Test
+    fun seconds() {
+        val tolerance = 1
+        val diff = abs(System.currentTimeMillis() / 1000 - Time.SystemImpl.seconds())
+        assertTrue(diff < tolerance)
+    }
+
+    @Test
+    fun millis() {
+        val tolerance = 100
+        val diff = abs(System.currentTimeMillis() - Time.SystemImpl.millis())
+        assertTrue(diff < tolerance)
+    }
+
+    @Test
+    fun nanoTime() {
+        val tolerance = 100000000
+        val diff = abs(System.nanoTime() - Time.SystemImpl.nanoTime())
+        assertTrue(diff < tolerance)
+    }
```

Comment:
I still don't understand purpose of those tests 🙂 
Do you think that something could go wrong in the implementation:
```kotin
override fun millis(): Long = System.currentTimeMillis()
```
?

Reply:
"Anything that can go wrong will go wrong". :)
I'm not saying these are very important tests. It's more of a habit: I try to test all my code when possible.
Idk, you can simply copy-paste something carelessly and end up with an implementation like:
```
override fun millis(): Long = System. nanoTime()
```
Again, not very probable but since it's just a unit test I think it can't hurt.

Reply:
agreeing with vadzim here. these tests are not needed.

It is also weird to make this a unit test, when they are really instrumentation tests. They are testing the time system calls. With unit tests, you mock deterministic values.

This unit test does not give any value, and it's essentially what the tests are doing if they are unit tests.
```
every { System.nanoTime } returns 100L
every { Time.SystemImpl.nanoTime() } returns 100L

assertEquals(System.nanoTime, Time.SystemImpl.nanoTime())
```

Reply:
If you want to unit test the logic of `SystemImpl`, it would look something like this

```
@Test
fun `nanoTime returns System.nanoTime`() {
  every { System.nanoTime() } returns 123

  assertEquals(123, SystemImpl.nanoTime())
}

@Test
fun `millis returns System.currentTimeMillis`() {
  every { SystemClock.currentTimeMillis() } returns 321

  assertEquals(321, SystemImpl.millis())
}

@Test
fun `seconds converts System.currentTimeMillis to seconds`() {
  every { SystemClock.currentTimeMillis() } returns 3999

  assertEquals(3, SystemImpl.seconds())
}
```

The tests make it more obvious, what to expect.

Reply:
Sure, that would be better but `System` class can't be mocked unfortunately. That's why I chose this approach. 
But if you are strongly against it, I'll remove the tests.

Reply:
There were reasons for us to prefer `SystemClock` over the `System` calls. They are actually different clocks and nav native uses `SystemClock` for monotonic location timestamps. Was there a reason you were preferring the `System` clock? I'm looking for discussions but found this comment https://github.com/mapbox/navigation-sdks/issues/636#issuecomment-691239489

Also, `SystemClock` has some existing tests that are mocking it https://github.com/mapbox/mapbox-navigation-android/blob/ce11135077d613bc821038b6bf945b38c6fd8549/libnavigation-core/src/test/java/com/mapbox/navigation/core/replay/history/MapboxReplayerTest.kt#L576-L592

Reply:
I didn't know there was a preference of `SystemClock` over `System`. Both can be encountered across the project code base. Here I only need it to check the time difference. There's absolutely no need to think it with NN.
I've switched to SystemClock elapsed time here. I think it makes sense.

Reply:
According to `SystemClock` docs the main difference between `System.nanoTime` and `SystemClock.elapsedRealtime(Nanos)` is that the latter includes time spent in power saving modes, so it may be more preferable.